src/test/java/eu/elixir/ega/ebi/it/KeyServiceITest.java

```diff
@@ -0,0 +1,102 @@
+package eu.elixir.ega.ebi.it;
+
+import static eu.elixir.ega.ebi.constants.Constants.KEY_FILEKEYS;
+import static eu.elixir.ega.ebi.constants.Constants.KEY_RETRIEVE;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+import org.json.JSONObject;
+import org.junit.Test;
+
+import com.jayway.restassured.response.Response;
+
+/**
+ * Class to do the integration test for KeyController in ega-data-api-key.
+ * 
+ * @author amohan
+ *
+ */
+public class KeyServiceITest extends KeyBase {
+
+    /**
+     * Verify the api call /keys/filekeys/{fileId} and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     */
+    @Test
+    public void testGetFileKey() {
+        final Response response = REQUEST.get(KEY_FILEKEYS + fileId);
+        response.then().assertThat().statusCode(SC_OK);
+        assertNotNull(response.body().asString());
+    }
+
+    /**
+     * Verify the api call /keys/retrieve/{keyId}/public and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     */
+    @Test
+    public void testGetPublicKeyFromPrivate() {
+        final Response response = REQUEST.get(KEY_RETRIEVE + keyId + "/public/");
+        response.then().assertThat().statusCode(SC_OK);
+        assertOutputKeyId(response.body().asString());

```

Comment:
In this and other tests you should check that the keyId that you get is the key that you expect. Otherwise if in some implementation there is a bug that gives you the wrong key, you won't catch it.

Reply:
It is actually checking the keyId, if you look inside the method call assertOutputKeyId(...)